build-image/first-boot.bash

```diff
@@ -20,44 +22,35 @@ fail_inprogress() {
 
 ###### start ######
 sleep 5
-echo -e "\n\n$(timestamp) [openHABian] Starting the openHABian initial setup."
+echo -e "$(timestamp) [openHABian] Starting the openHABian initial setup."
 rm -f /opt/openHABian-install-failed
 touch /opt/openHABian-install-inprogress
 
 echo -n "$(timestamp) [openHABian] Storing configuration... "
-cp /boot/openhabian.conf "$CONFIGFILE"
-sed -i 's/\r$//' "$CONFIGFILE"
-
-# shellcheck disable=SC1090
-source "$CONFIGFILE"
-# shellcheck disable=SC1091
-source "/boot/helpers.bash"
-# shellcheck disable=SC1091
-source "/boot/openhabian.bash"
-echo "OK"
+if ! cp /boot/openhabian.conf "$CONFIGFILE"; then echo "FAILED (copy)"; fail_inprogress; fi
+if ! sed -i 's|\r$||' "$CONFIGFILE"; then echo "FAILED (Unix line endings)"; fail_inprogress; fi
+if ! source "$CONFIGFILE"; then echo "FAILED (source config)"; fail_inprogress; fi
+if ! source "/boot/helpers.bash"; then echo "FAILED (source helpers)"; fail_inprogress; fi
```

Comment:
The copy paste is gone, I think that we just re-source them after updating because they are needed before updating

Reply:
Actually, I am not sure how you plan on doing that, As the functions are going to be the same actions are used across both places? Let me try and change it and see if that is what you want.

Reply:
So far we copy those 2 files to /boot (build.bash L290f). Now change to source /opt/openhabian/functions/x instead and remove the copy but on image build, add to git-clone all of openhabian repo to /opt/openhabian because that needs to be there anyway for offline install to work.

Reply:
Better?

Reply:
Sorry, GitHub updated slow, WIP

Reply:
Like that?

Reply:
yes ok